### PR TITLE
feat: Create ToggleIconButton primitive for rich-text-editor

### PR DESCRIPTION
**Pull request by dougmacknz**
# Objective
Creates a new component called `ToggleIconButton` inside a newly created `@kaizen/rich-text-editor` package.

# Motivation and Context
This component will be one type of `ToolbarItem` that could be included in a Rich Text Editor `Toolbar`. It will be the only one we create for now, for use with bold, italics, underline, and list functionality.

### CODE REVIEW

File: yarn.lock
Comment (HeartSquared) [inline yarn.lock:16989]: This is strange

File: packages/rich-text-editor/src/components/ToggleIconButton/ToggleIconButton.scss
Comment (HeartSquared) [inline packages/rich-text-editor/src/components/ToggleIconButton/ToggleIconButton.scss:6]: Do we need to change the cursor? I noticed that Slack's RTE buttons use a pointer

**Review by HeartSquared — CHANGES_REQUESTED** [review]
Mostly questions and clean up - overall it looks fine.
Also the `index.ts` is empty.

File: packages/rich-text-editor/docs/ToggleIconButton.stories.tsx
Comment (HeartSquared) [inline packages/rich-text-editor/docs/ToggleIconButton.stories.tsx:31]: Not sure if you've tested other icons, but might be worth including the relevant ones in a story to ensure they all render correctly (eg. alignment next to each other, whether the current styles work with all icons or if it looks off when changed).

File: packages/rich-text-editor/src/components/ToggleIconButton/ToggleIconButton.scss
Comment (dougmacknz) [reply packages/rich-text-editor/src/components/ToggleIconButton/ToggleIconButton.scss:19]: It's one of the questions we have for the designers at the moment, waiting for hear back.

File: packages/rich-text-editor/src/components/ToggleIconButton/ToggleIconButton.scss
Comment (dougmacknz) [reply packages/rich-text-editor/src/components/ToggleIconButton/ToggleIconButton.scss:6]: I'd defer to https://www.w3.org/TR/wai-aria-practices/examples/toolbar/toolbar.html and just the general fact that browsers don't put pointers on buttons.

Related: https://adamsilver.io/blog/buttons-shouldnt-have-a-hand-cursor/

File: packages/rich-text-editor/docs/ToggleIconButton.stories.tsx
Comment (mcwinter07) [reply packages/rich-text-editor/docs/ToggleIconButton.stories.tsx:28]: Might hold off on producing more stories for this individual component for now until we can the Toolbar and ToolbarSection (also design feedback on the button itself)

File: packages/rich-text-editor/src/components/ToggleIconButton/ToggleIconButton.scss
Comment (HeartSquared) [reply packages/rich-text-editor/src/components/ToggleIconButton/ToggleIconButton.scss:6]: Interesting. Especially since it's much more common to see the pointer on a button on websites than not. I wonder how long it will take for people to follow the convention, considering that the more common pattern is actually to have the pointer (as he also mentioned he associated it himself).

I just read part 2 of the blog too, but I still have a question.

In the case where we have an `<a>` that looks like a button, do we still have the pointer? What about the opposite case where `<button>` looks like a link?

File: packages/rich-text-editor/src/components/ToggleIconButton/ToggleIconButton.tsx
Comment (HeartSquared) [inline packages/rich-text-editor/src/components/ToggleIconButton/ToggleIconButton.tsx:37]: Oops forgot in my first review.

```suggestion
  })
  
ToggleIconButton.displayName = "ToggleIconButton"
```

File: packages/rich-text-editor/src/components/ToggleIconButton/ToggleIconButton.tsx
Comment (HeartSquared) [inline packages/rich-text-editor/src/components/ToggleIconButton/ToggleIconButton.tsx:17]: Just saw the file structure too. Looking at our docs, it should be:
`src/RichTextEditor/components/ToggleIconButton`
as the `components` folder is only necessary for when you have subcomponents.

File: packages/rich-text-editor/src/components/ToggleIconButton/ToggleIconButton.scss
Comment (mcwinter07) [reply packages/rich-text-editor/src/components/ToggleIconButton/ToggleIconButton.scss:6]: In theory, if it does act as a link I would assume so? Though I think that A11y best practice is to have your links visually distinct from your buttons - I'm not 100% sure if it would directly violate a rule (ie: not having your links visually distinct from text) but ideally links have unique styles that are consistent :/

File: packages/rich-text-editor/src/components/ToggleIconButton/ToggleIconButton.tsx
Comment (mcwinter07) [reply packages/rich-text-editor/src/components/ToggleIconButton/ToggleIconButton.tsx:17]: Mind if I get a sense check from the both of you @HeartSquared @dougmacknz before I push this change:

Screen Shot 2022-03-04 at 3 35 02 pm

File: packages/rich-text-editor/src/components/ToggleIconButton/ToggleIconButton.scss
Comment (dougmacknz) [reply packages/rich-text-editor/src/components/ToggleIconButton/ToggleIconButton.scss]: Looks like our Button component currently does a pointer regardless, I'd definitely be in favour of changing that.

The button that looks like a link situation feels a bit more icky, but it's not a case we've run into yet (that I'm aware of anyway).

Related: https://github.com/cultureamp/kaizen-design-system/discussions/1868

File: packages/rich-text-editor/src/components/ToggleIconButton/ToggleIconButton.tsx
Comment (HeartSquared) [reply packages/rich-text-editor/src/components/ToggleIconButton/ToggleIconButton.tsx]: Yep that looks right! :)

**Review by HeartSquared — APPROVED** [review]
LGTM :D

**Pull request merged**
